### PR TITLE
W-11584645: Add a mechanism to ignore levels in hierarchy of spans

### PR DESCRIPTION
**Pull request by fsgonz**

### CODE REVIEW

**Comment by fsgonz** [comment]
--validate

**Review by IvanAndresFritzler — CHANGES_REQUESTED** [review]

File: core-tests/src/test/java/org/mule/runtime/core/internal/profiling/consumer/tracing/DefaultCoreEventTracerTestCase.java [hunk 1]
```diff
@@ -113,14 +117,42 @@ public class DefaultCoreEventTracerTestCase extends AbstractMuleTestCase {
...
+      public ChildSpanCustomizationInfo getChildSpanCustomizationInfo() {
+        return getDefaultChildSpanInfo();
+      }
+    }, componentIdentifier);
+
+    assertThat(span.getName(), equalTo(getSpanName(componentIdentifier)));
+  }
+
+  @Test
+  public void testStartComponentExecution2() {
```
[the remark this answers is not in the document]
Comment (fsgonz) [reply core-tests/src/test/java/org/mule/runtime/core/internal/profiling/consumer/tracing/DefaultCoreEventTracerTestCase.java:138]: yes 😱

**Review by IvanAndresFritzler — DISMISSED** [review]

**Review by IvanAndresFritzler — APPROVED** [review]

**Review by IvanAndresFritzler — DISMISSED** [review]

File: core/src/main/java/org/mule/runtime/core/internal/profiling/tracing/export/InternalSpanExporterVisitor.java [hunk 2]
```diff
@@ -20,4 +22,9 @@
    */
   T accept(OpenTelemetrySpanExporter opentelemetrySpanExporter);
 
+  /**
+   * @param opentelemetrySpanExporter the {@link NoOpInternalSpanExporter} to accept
+   * @return the result of visiting the instnce.
```
Comment (sofiamorseletto) [inline core/src/main/java/org/mule/runtime/core/internal/profiling/tracing/export/InternalSpanExporterVisitor.java:27]: instance*

File: core/src/main/java/org/mule/runtime/core/internal/profiling/tracing/event/span/export/optel/OpentelemetrySpanExporterManager.java [hunk 3]
```diff
@@ -23,8 +23,8 @@
  */
 public class OpentelemetrySpanExporterManager implements InternalSpanExportManager<EventContext> {
 
-  private static final OpenetelemetryCoreEventInternalSpanExporterFactory SPAN_EXPORTER_FACTORY =
-      getOpenetelemetryCoreEventInternalSpanExporterFactory();
+  private static final OpentelemetryCoreEventInternalSpanExporterFactory SPAN_EXPORTER_FACTORY =
```
Comment (sofiamorseletto) [inline core/src/main/java/org/mule/runtime/core/internal/profiling/tracing/event/span/export/optel/OpentelemetrySpanExporterManager.java:26]: Shouldn't it be OpenTelemetry... ? Both here as in getOpentelemetryCoreEventInternalSpanExporterFactory

**Pull request merged**